mmf/modules/bottleneck.py

```diff
@@ -0,0 +1,228 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+
+from typing import Optional, Tuple, Type
+
+import torch
+import torch.nn as nn
+from torchvision.models.resnet import conv1x1, conv3x3
+
+
+class ChannelPool(nn.Module):
+    """Average pooling in the channel dimension"""
+
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        return x.mean(dim=1, keepdim=True)
+
+
+class SEModule(nn.Module):
+    """Squeeze-and-Excitation module from https://arxiv.org/pdf/1709.01507.pdf
+
+    Args:
+        dim: the original hidden dim.
+        sqrate: the squeeze rate in hidden dim.
+    Returns:
+        New features map that channels are gated
+        by sigmoid weights from SE module.
+    """
+
+    def __init__(self, dim: int, sqrate: float):
+        super().__init__()
+        self.se = nn.Sequential(
+            nn.AdaptiveAvgPool2d((1, 1)),
+            nn.Conv2d(dim, dim // sqrate, kernel_size=1, bias=False),
+            nn.ReLU(inplace=True),
+            nn.Conv2d(dim // sqrate, dim, kernel_size=1, bias=False),
+            nn.Sigmoid(),
+        )
+        self.attn = nn.Sequential(
+            ChannelPool(),
+            nn.Conv2d(1, 1, kernel_size=7, padding=3, bias=False),
+            nn.Sigmoid(),
+        )
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = x * self.se(x)
+
+        return x * self.attn(x)
+
+
+class FrozenBatchNorm2d(nn.Module):
```

Comment:
Please see https://github.com/pytorch/vision/blob/39021408587eb252ebb842a54195d840d6d76095/torchvision/ops/misc.py#L43

Reply:
Might make sense to inherit it and call it `MovieMCANFrozenBatchNorm2d`